packages/common/src/SolidityTestUtils.ts

```diff
@@ -136,10 +138,10 @@ export async function mineTransactionsAtTime(
 
       // Awaits the transactionHash, which signifies the transaction was sent, but not necessarily mined.
       await new Promise<void>((resolve, reject) => {
-        result.on("transactionHash", function () {
+        result.once("transactionHash", function () {
           resolve();
```

Comment:
just one note here, this is resolving with no hash, hopefully callers dont expect the hash value from the receipts, it will always be an empty array

Reply:
@daywiss callers _do not_ expect this. Note the await in front of the promise -- this promise's result is never being used, so we just want to wait until the txn hash has been generated, we don't actually need to use it. But good call out!